explore_data/cluster_by_accessory_gene.py

```diff
@@ -0,0 +1,609 @@
+print(f"Number of PA14 samples if {num_pa14_samples}")
+print("\n\n")
+
+# Of those samples, how many have o or low expression?
+threshold = 50
+pao1_sample_ids = pao1_pa14_acc_expression_label["Strain type_pao1"] == "PAO1"
+pao1_acc_expression_label = pao1_pa14_acc_expression_label.loc[pao1_sample_ids]
+pao1_core_expression_label = pao1_pa14_core_expression_label.loc[pao1_sample_ids]
+
+print("----- Matching sample and reference statistics -------")
+pao1_acc_expression_low = pao1_acc_expression_label[
+    pao1_acc_expression_label["median acc expression_pao1"] == 0
+]
+print(
+    f"Proportion of PAO1 samples with 0 median PAO1-only expression: {pao1_acc_expression_low.shape[0]/num_pao1_samples}"
+)
+pao1_acc_expression_low = pao1_acc_expression_label[
+    pao1_acc_expression_label["median acc expression_pao1"] < threshold
+]
+print(
+    f"Proportion of PAO1 samples with median PAO1-only expression < {threshold}: {pao1_acc_expression_low.shape[0]/num_pao1_samples}"
+)
+pao1_core_expression_low = pao1_core_expression_label[
+    pao1_core_expression_label["median core expression_pao1"] < threshold
+]
+print(
+    f"Proportion of PAO1 samples with median core expression < {threshold}: {pao1_core_expression_low.shape[0]/num_pao1_samples}"
+)
+
+pa14_sample_ids = pao1_pa14_acc_expression_label["Strain type_pao1"] == "PA14"
+pa14_acc_expression_label = pao1_pa14_acc_expression_label.loc[pa14_sample_ids]
+pa14_core_expression_label = pao1_pa14_core_expression_label.loc[pa14_sample_ids]
+
+pa14_acc_expression_low = pa14_acc_expression_label[
+    pa14_acc_expression_label["median acc expression_pa14"] == 0
+]
+print(
+    f"Proportion of PA14 samples with 0 median PA14-only expression: {pa14_acc_expression_low.shape[0]/num_pa14_samples}"
+)
+pa14_acc_expression_low = pa14_acc_expression_label[
+    pa14_acc_expression_label["median acc expression_pa14"] < threshold
+]
+print(
+    f"Proportion of PA14 samples with median PA14-only expression < {threshold}: {pa14_acc_expression_low.shape[0]/num_pa14_samples}"
+)
+pa14_core_expression_low = pa14_core_expression_label[
+    pa14_core_expression_label["median core expression_pa14"] < threshold
+]
+print(
+    f"Proportion of PA14 samples with median core expression < {threshold}: {pa14_core_expression_low.shape[0]/num_pa14_samples}"
+)
+print("\n\n")
+
+print("----- mis-matching sample and reference statistics -------")
+pa14_acc_expression_low = pa14_acc_expression_label[
+    pa14_acc_expression_label["median acc expression_pao1"] == 0
+]
+print(
+    f"Proportion of PA14 samples with 0 median PAO1-only expression: {pa14_acc_expression_low.shape[0]/num_pa14_samples}"
+)
+pao1_acc_expression_low = pao1_acc_expression_label[
+    pao1_acc_expression_label["median acc expression_pa14"] == 0
+]
+print(
+    f"Proportion of PAO1 samples with 0 median PA14-only expression: {pao1_acc_expression_low.shape[0]/num_pao1_samples}"
+)
+# -
+
```

Comment:
Is there a way to make the above results into a confusion matrix?

Reply:
This is a great suggestion. I think this type of visualization will help make the interpretation easier.